salesforce/10.10/modules/ROOT/pages/salesforce-connector-processing-events.adoc

```diff
@@ -9,88 +9,90 @@ You can use Anypoint Connector for Salesforce (Salesforce Connector) as an inbou
 
 == Prerequisites
 
-To process events with Salesforce Connector, ensure you meet the following prerequisites:
+To process events with Salesforce Connector, ensure that you have:
+
+* Access to the Salesforce target resource
+* The required Salesforce Streaming API permissions enabled in your organization
 
-* You must have access to the Salesforce target resource.
-* Before you can receive events for Salesforce changes associated with a topic, you must create the topic, if it does not exist.
-* You must have the required Salesforce Streaming API permissions enabled in your organization.
+To receive events for Salesforce changes associated with a topic, create a new topic or use an existing one.
 
 [[objectstoreusage]]
-== Object Store Usage
+== Object Store Usage in Events
 
-Both Salesforce Connector and Mule use an object store to persist data for features such as automatic message replay and message redelivery:
+Both Salesforce Connector and Mule runtime engine (Mule) use an object store to persist data for features such as automatic message replay and message redelivery:
 
 * The object store included in Mule for on-premises deployments of Mule apps has no transaction limits.
 * The free version of Object Store for CloudHub deployments of Mule apps has a limit of 10 transactions per second.
 
-For more information about object store versions, see https://docs.mulesoft.com/object-store/#object-store-notes[Object Store Notes].
-
+For more information about Object Store versions, see https://docs.mulesoft.com/object-store/#object-store-notes[Object Store Notes].
 
 === Replay Topic and Replay Channel Listener Operations
 
-The *Replay topic* and *Replay channel listener* operations have the option to continue from the last replay ID they received before restarting the application.
+The *Replay topic* and *Replay channel listener* operations can continue from the last replay ID they received before restarting the application.
 
-When a Mule app starts for the first time, the connector creates an object store that is used to save data related to messages that were processed successfully or that failed:
+When a Mule app starts for the first time, the connector creates an object store that saves data:
 
-* For each message that was processed successfully, the information stored consists of a replay ID associated with the message.
-* For each message that failed, the information stored consists of a number that represents the lowest replay ID for which the processing failed.
+* For each message that processes successfully, the connector stores the replay ID associated with the message.
+* For each message that fails, the connector stores a number that represents the lowest replay ID for which the processing failed.
```

Comment:
"a number..." versus the actual replay ID, as implied by the peer list item?
```suggestion
* For each message that fails, the connector stores the lowest replay ID for which the processing failed.
```
I'm confused: if something fails, it fails once, at the first instance that it cannot process. Isn't this automatically the lowest/first ID for which it fails: do we need to say "lowest"? Are there other failures after the first?

Reply:
There are 2 things: the order in which the events are received from Salesforce. And second: the order in which the events are processed by the Mule flow that is right next to the source. The evets come from Salesforce in incremental order. For example: 2 3 4 5 7 8 9 11. Now, the source might process them like this: 11 3 4 5 2 7 8 9.
Let's suppose event with replayId = 3 fails. Then we save as lowest failed replay id the number 3. Then, 4 and 5 are processed successfully. But event with replayId = 2 fails. Now the lowest failed replay id that we save is: 2. 
If the source is now restarted and it has the "Resume from last replay id" option selected, then it will start from 2.

Reply:
The actual replay id is saved. I do not think the user needs to know what we save internally. Only thing they need to know is that the restart logic of a source that has the "Resume from last replay id" option selected is:
1) If there are one or more failed events, the source will start with the lowest replay id that failed;
2) If there are no failed events, the source will start from the last successfully processed event id.